.github/workflows/backport.yaml

```diff
@@ -0,0 +1,46 @@
+name: Pull Request Backporting using Git Backporting
+
+on:
+  pull_request_target:
+    types:
+      - closed
+      - labeled
+
+env:
+  GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+  NO_SQUASH_OPTION: true
+
+jobs:
+  backporting:
+    name: "Backporting"
+    concurrency:
+        group: backporting-${{ github.head_ref }}
+        cancel-in-progress: true
+    # Only react to merged PRs for security reasons.
+    # See https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request_target.
+    if: >
+      github.event.pull_request.merged
+      && (
+        github.event.action == 'closed'
+          && (contains(github.event.pull_request.labels.*.name, 'backport')
+            || contains(github.event.pull_request.labels.*.name, 'backport-squash'))
+        || (
+          github.event.action == 'labeled'
+          && contains(github.event.label.name, 'backport')
+        )
+      )
+    runs-on: ubuntu-latest
+    steps:
+      - name: Override no-squash option
+        if: >
+          (github.event.action == 'closed' && contains(github.event.pull_request.labels.*.name, 'backport-squash'))
+          || (github.event.action == 'labeled' && contains(github.event.label.name, 'backport-squash'))
```

Comment:
That's true, but if I check `github.event.pull_request.labels.*.name` when adding a label I am actually checking if at least one label with that name exists, which is true even if I am adding a new label and `backport` already exists on that pul request. This is not what we want as it would trigger the workflow multiple times.

Reply:
Good point...but, I think that applies to my previous suggestion (which I shall withdraw 🙂).

In this spot, we've already decided to run the workflow, so the only question is whether to squash or not, and for that I think you can check `github.event.pull_request.labels.*.name` in either case.

Reply:
See other comment https://github.com/Hyperfoil/horreum-client-python/pull/4#discussion_r1539800301 I don't think that is enough to check `backport` word. Anyway we are talking about minors so not a big deal